[Linux] Fix Lintian warnings

basilgello opened the pull request:
## Description

Fixes Lintian warning.

## Motivation and Context

Trying to make Debian pakcage lintian-clean :+1: 

## How Has This Been Tested?

Build; run lintian on resulting deb-files.

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [x] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [x] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [x] All new and existing tests passed

basilgello committed:
[Linux] Add keywords to desktop files

Fixes Lintian warning.

basilgello commented:
@phunkyfish

mapreri commented:
More references in https://specifications.freedesktop.org/desktop-entry-spec/latest/ar01s06.html and https://specifications.freedesktop.org/desktop-entry-spec/latest/apes04.html

basilgello committed:
[Python] Replace shebang with 'env python3' invocation

This is conformant to POSIX and should work on Windows and OSX too.
Also this fixes Lintian 'unusual-interpreter' warnings.

basilgello commented:
I am going to add several other commits fixing Lintian warnings. All them are safe to include before 19.0 beta1 as they are cosmetic changes.

garbear reviewed (approved):
No risky code changes; Python scripts aren't executed from command line, so the shebang doesn't matter (what would this do on Android or iOS?); and the desktop shortcut changes are welcome.

basilgello committed:
Add TexturePacker manpage

fuzzard commented on tools/darwin/Support/GenerateMissingImages-tvos.py:
```diff
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/env python3
```
Do not change this. Does not work on OSX Catalina

```
Support % ./GenerateMissingImages-tvos.py ../../../xbmc/platform/darwin/tvos/Assets.xcassets Assets LaunchImage
/Users/brent/osx-master/media/splash.jpg
  /Users/brent/osx-master/xbmc/platform/darwin/tvos/Assets.xcassets/LaunchImage.launchimage/splash@2x.jpg
/Users/brent/osx-master/xbmc/platform/darwin/tvos/Assets.xcassets/Assets.brandassets/topshelf_wide.imageset/image.png
  /Users/brent/osx-master/xbmc/platform/darwin/tvos/Assets.xcassets/Assets.brandassets/topshelf_wide.imageset/image@2x.png
Traceback (most recent call last):
  File "./GenerateMissingImages-tvos.py", line 28, in <module>
    for i in xrange(1, 5):
NameError: name 'xrange' is not defined
```

basilgello replied:
Thanks for testing this! Yes xrange changes is trivial - I will walk through the files and fix the py2-specific issues.

basilgello replied:
I fixed this issue. Please test it when you can.

fuzzard replied:
this change is ok now.

fuzzard commented:
Have any of the py changes been tested running on py3? 
There are changes in the language, and most look like they were definitely built in the py2 era. Some may need language changes, and not just s shebang change.

basilgello commented:
I do use kodi-send sinve I migrated to 19.0 in April. As Python 2 gets deprecated anyway, I think I will fix the scripts to work with Python 3 because niw it is an ideal moment for it.
-- 
Vasyl Gello
==================================================
Certified SolidWorks Expert

Mob.:+380 (98) 465 66 77

E-Mail: vasek.gello@gmail.com

Skype: vasek.gello
==================================================
호랑이는 죽어서 가죽을 남기고 사람은 죽어서 이름을 남긴다

garbear commented on docs/manpages/TexturePacker.1:
```diff
@@ -0,0 +1,24 @@
+.BR \-dupecheck
+Check for image duplicates first
+.TP
+.BR \-inout
```
`input` I believe

basilgello replied:
Fixed.

basilgello commented:
Now I am going to run-test tge affected python scripts once more and add another commit fixing manpage for `kodi.bin`

basilgello committed:
Port zeroconf.py to Python 3

basilgello committed:
[EventClients] Fix examples and libs

 * Provide correct icon paths and types depending on file existence
 * Fix C++ and Java EventClient examples

basilgello committed:
Refresh manpages

basilgello commented:
OK, this PR is ready for review.

garbear reviewed (approved):
Some code changes have been added, but they're all in the `tools/` folder so I have no concerns.

basilgello commented:
@fuzzard can you please verify the tvOS helper works as expected? The examples except C# have been verified by me. PS3 scripts also start and try pairing with BT joystick but I dont have one.

fuzzard commented on addons/service.xbmc.versioncheck/resources/lib/version_check/service.py:
```diff
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/env python3
```
This change has to be done upstream https://github.com/XBMC-Addons/service.xbmc.versioncheck

enen92 replied:
Done and merged

fuzzard commented on addons/service.xbmc.versioncheck/resources/lib/version_check/viewer.py:
```diff
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/env python3
```
same again https://github.com/XBMC-Addons/service.xbmc.versioncheck

fuzzard reviewed (commented):
the demo metadata changes are ok in tree, but the service.xbmc.versioncheck changes needs to be sent upstream, otherwise they will just be wiped next time its synced.

wsnipex commented:
jenkins build this please

DaveTBlake commented:
Have the versioncheck changes been submitted upstream? And if so do they need to be here too since repo sync will overwrite.

basilgello commented:
Not yet, sorry @DaveTBlake! Will do tonight!

DaveTBlake commented:
Going to bump to next release to ensure so that version check changes get done in right place. Looking to merge this soonest though @basilgello

basilgello commented:
@DaveTBlake @phunkyfish

phunkyfish reviewed (approved):
Thanks for upstreaming. Looks good now.

howie-f commented:
jenkins build this please

howie-f commented:
jenkins build this please

howie-f commented:
jenkins build this please

howie-f commented:
jenkins build this please
